### PR TITLE
toggle fade of other lines when selecting a point for longitudinal co…

### PR DESCRIPTION
**Pull request by dsnyder-fairway**
…hort chart

screen shot 2018-05-07 at 7 14 03 pm

screen shot 2018-05-07 at 7 13 55 pm

### CODE REVIEW

**Comment by dsnyder-fairway** [comment]
I'll re-open this after resolving conflicts tomorrow

File: webapp/src/main/webapp/src/app/aggregate-report/results/longitudinal-cohort-chart.component.html [hunk 1]
```diff
@@ -51,7 +51,7 @@
            [ngClass]="linePallet">
 
           <g *ngFor="let performancePath of chartView.performancePaths; let i = index"
-             [ngClass]="performancePath.visible ? '' : ' hidden'">
+             [ngClass]="performancePath.fade ? '' : ' fade'">
```
Comment (jtreuting) [inline webapp/src/main/webapp/src/app/aggregate-report/results/longitudinal-cohort-chart.component.html]: nit: that extra space in the class name shouldn't be needed

Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/aggregate-report/results/longitudinal-cohort-chart.component.html]: corrected this

File: webapp/src/main/webapp/src/app/aggregate-report/results/longitudinal-cohort-chart.component.ts [hunk 2]
```diff
@@ -162,7 +163,23 @@ export class LongitudinalCohortChartComponent implements OnInit {
       : this._selectedPaths.add(pathIndex);
 
     this._chartView.performancePaths
-      .forEach((path, index) => path.visible = this._selectedPaths.size === 0 || this._selectedPaths.has(index));
+      .forEach((path, index) => path.fade = this._selectedPaths.size !== 0 && !this._selectedPaths.has(index));
+    this.previousPoint = null;
```
[the remark this answers is not in the document]
Comment (dsnyder-fairway) [reply webapp/src/main/webapp/src/app/aggregate-report/results/longitudinal-cohort-chart.component.ts]: when switching clicking a point on the same line onPopoverHide would be called. If we pass in the argument point as well it appears to call the onPopoverHide with whatever point was selected before. I might be missing the for-loop should be placed...

[the remark this answers is not in the document]
Comment (esotrope) [reply webapp/src/main/webapp/src/app/aggregate-report/results/longitudinal-cohort-chart.component.ts]: yeah that sounds odd : / would take some more tinkering to change i suppose

**Pull request merged**